src/python/pants/base/exception_sink.py

```diff
@@ -478,11 +476,19 @@ def _log_unhandled_exception_and_exit(cls, exc_class=None, exc=None, tb=None, ad
 
     # Generate an unhandled exception report fit to be printed to the terminal (respecting the
     # Exiter's should_print_backtrace field).
-    stderr_printed_error = cls._format_unhandled_exception_log(
-      exc, tb, add_newline,
-      should_print_backtrace=cls._should_print_backtrace_to_terminal)
-    if extra_err_msg:
-      stderr_printed_error = '{}\n{}'.format(stderr_printed_error, extra_err_msg)
+    if cls._should_print_backtrace_to_terminal:
+      stderr_printed_error = cls._format_unhandled_exception_log(
+        exc, tb, add_newline,
+        should_print_backtrace=cls._should_print_backtrace_to_terminal)
+      if extra_err_msg:
+        stderr_printed_error = '{}\n{}'.format(stderr_printed_error, extra_err_msg)
+    else:
+      # If the user didn't ask for a backtrace, show a succinct error message without
+      # all the exception-related preamble.  A power-user/pants developer can still
+      # get all the preamble info along with the backtrace, but the end user shouldn't
+      # see that boilerplate by default.
+      error_msgs = getattr(exc, 'end_user_messages', lambda: [str(exc)])()
```

Comment:
`repr(exc)` is even better, because it will ensure the type of exception is printed like `ValueError`. For example, when doing f-strings with exceptions in `except` blocks it's best to do `{e!r}` rather than `{e}`.